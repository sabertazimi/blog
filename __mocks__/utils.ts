let navigatorClipboardMocked = false
const writeTextMock = jest.fn()

const mockNavigatorClipboard = () => {
  if (!navigatorClipboardMocked) {
    Object.defineProperty(global.navigator, 'clipboard', {
      value: {
        writeText: writeTextMock,
      },
    })

    navigatorClipboardMocked = true
  }

  return { writeTextMock }
}

export { mockNavigatorClipboard }
