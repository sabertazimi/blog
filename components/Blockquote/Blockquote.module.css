.blockquote {
  @apply mx-0 mb-9 rounded-lg border-l-10 border-solid border-primary bg-gray-100 px-6 pb-6 pt-12;

  quotes: '\201C''\201D';
}

.blockquote::before {
  @apply mr-2 align-bottom text-9xl leading-12 text-primary;

  content: open-quote;
}

.blockquote div,
.blockquote p {
  @apply inline italic;
}

/* Keep nested blockquote keep compacted */
.blockquote blockquote {
  @apply mb-3 ml-2;
}
