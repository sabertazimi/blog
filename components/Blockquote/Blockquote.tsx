import cx from 'classnames'
import type { BlockquoteHTMLAttributes } from 'react'
import styles from './Blockquote.module.css'

interface Props extends BlockquoteHTMLAttributes<HTMLElement> {}

function Blockquote({ children, className, ...props }: Props): JSX.Element {
  return (
    <blockquote
      {...props}
      className={cx(
        className,
        styles.blockquote,
        'dark:border dark:bg-black dark:shadow-xl dark:shadow-primary',
      )}
    >
      {children}
    </blockquote>
  )
}

export default Blockquote
