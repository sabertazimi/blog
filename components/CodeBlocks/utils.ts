import parseNumericRange from 'parse-numeric-range'
import type { Language } from 'prism-react-renderer'

const normalizeCode = (code: string = '') => code.replace(/\n+$/, '')

function isLanguage(language: string): language is Language {
  return language.length > 0
}

function normalizeLanguage(languageClass?: string): Language {
  if (languageClass != null) {
    const language = languageClass.replace('language-', '')

    if (isLanguage(language))
      return language
  }

  return 'typescript'
}

function normalizeLanguageName(language: string): string {
  switch (language) {
    case 'html':
      return 'HTML'
    case 'xml':
      return 'XML'
    case 'yml':
    case 'yaml':
      return 'YAML'
    case 'css':
      return 'CSS'
    case 'json':
      return 'JSON'
    case 'md':
    case 'markdown':
      return 'Markdown'
    case 'js':
    case 'javascript':
      return 'JavaScript'
    case 'ts':
    case 'typescript':
      return 'TypeScript'
    case 'coffee':
    case 'coffeescript':
      return 'CoffeeScript'
    case 'jsx':
    case 'tsx':
      return 'React'
    case 'objc':
    case 'objectivec':
      return 'Objective-C'
    default:
      return language.charAt(0).toUpperCase() + language.slice(1)
  }
}

function normalizeLines(expression: string): Set<number> {
  return new Set(parseNumericRange(expression))
}

export {
  normalizeCode,
  normalizeLanguage,
  normalizeLanguageName,
  normalizeLines,
}
