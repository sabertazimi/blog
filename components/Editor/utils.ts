function languageToFilepath(language?: string): string {
  switch (language) {
    case 'css':
      return '/styles.css'
    case 'js':
    case 'javascript':
      return '/App.js'
    case 'ts':
    case 'typescript':
      return '/App.ts'
    case 'jsx':
      return '/App.jsx'
    case 'tsx':
      return '/App.tsx'
    case 'vue':
      return '/src/App.vue'
    case undefined:
    default:
      return '/App.tsx'
  }
}

function normalizeFilepath(filename?: string, language?: string): string {
  if (filename == null || filename === '')
    return languageToFilepath(language)

  if (!filename.startsWith('/'))
    return `/${filename}`

  return filename
}

export { normalizeFilepath }
