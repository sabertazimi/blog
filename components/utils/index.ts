import classNames from 'classnames';

export { classNames };
