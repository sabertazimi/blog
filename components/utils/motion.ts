export { AnimatePresence, motion } from 'framer-motion';
export type { MotionProps } from 'framer-motion';
