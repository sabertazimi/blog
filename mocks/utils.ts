let navigatorClipboardMocked = false
const writeTextMock = jest.fn()

function mockNavigatorClipboard() {
  if (!navigatorClipboardMocked) {
    Object.defineProperty(globalThis.navigator, 'clipboard', {
      value: {
        writeText: writeTextMock,
      },
    })

    navigatorClipboardMocked = true
  }

  return { writeTextMock }
}

export { mockNavigatorClipboard }
