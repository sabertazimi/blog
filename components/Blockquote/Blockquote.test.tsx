import Blockquote from './Blockquote'
import { render } from '@/utils'

describe('Blockquote', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(<Blockquote>Blockquote</Blockquote>)

    expect(container).toMatchSnapshot()
  })
})
