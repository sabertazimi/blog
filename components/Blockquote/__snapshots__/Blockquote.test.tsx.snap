// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote should render correctly (snapshot) 1`] = `
<div>
  <blockquote
    class="blockquote dark:border dark:bg-black dark:shadow-xl dark:shadow-primary"
  >
    Blockquote
  </blockquote>
</div>
`;
